File: classes/parte-7-paradigmas/114-funcional-i-inmutabilidad-y-funciones-puras/implementaciones/typescript/main.ts
import { readFileSync } from "node:fs";

const nums: number[] = readFileSync(0, "utf8").trim().split(/\s+/).map(Number);
const doblados: number[] = nums.map((x) => x * 2);
console.log(`doblados=${doblados.join("-")}`);

File: classes/parte-7-paradigmas/114-funcional-i-inmutabilidad-y-funciones-puras/implementaciones/java/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] p = br.readLine().trim().split("\\s+");
        String r = Arrays.stream(p).map(Integer::parseInt).map(x -> x * 2)
                .map(String::valueOf).collect(Collectors.joining("-"));
        System.out.println("doblados=" + r);
    }
}
